docs/developer/decision-records/2024-10-09-vault-authentication-refactor/README.md

```diff
@@ -0,0 +1,72 @@
+# Decision Record Refactor of the HashiCorp Vault extension
+
+## Decision
+
+Refactor the HashiCorp Vault extension and make it extensible with different methods of authentication.
+
+## Rationale
+
+The current implementation of the authentication in the HashiCorp Vault extension has no way to add new authentication methods and to switch between them.
+It goes against the EDC extensibility model and needs to be remedied.
+Additionally, extracting the current implementation of the token authentication into its own extension will improve readability and maintainability of the code.
+
+## Affected Areas
```

Comment:
no need for this chapter, it can be described in the `Approach` one with one sentence like: 
the refactor will affect the `vault-hashicorp` extension